From 08704baaae0036b225824bb3412047199cb97bbb Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Tue, 6 Mar 2018 17:49:41 -0800
Subject: [PATCH 0321/1852] v4l: xilinx: dma: Handle 10 bit format calucations
 for XV20

commit 32cd8832e7d258b68fb1d5307c7a7d9064461f04 from
https://github.com/Xilinx/linux-xlnx.git

This patch handle 10 bit calculations for YUV 422 10 bit format

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.c b/drivers/media/platform/xilinx/xilinx-vip.c
index 1959aedbe857..002414aa095f 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.c
+++ b/drivers/media/platform/xilinx/xilinx-vip.c
@@ -126,6 +126,7 @@ void xvip_bpl_scaling_factor(u32 fourcc, u32 *numerator, u32 *denominator)
 {
 	switch (fourcc) {
 	case V4L2_PIX_FMT_XV15M:
+	case V4L2_PIX_FMT_XV20M:
 		*numerator = 10;
 		*denominator = 8;
 		break;
@@ -149,6 +150,7 @@ void xvip_width_padding_factor(u32 fourcc, u32 *numerator, u32 *denominator)
 {
 	switch (fourcc) {
 	case V4L2_PIX_FMT_XV15M:
+	case V4L2_PIX_FMT_XV20M:
 		/* 32 bits are required per 30 bits of data */
 		*numerator = 32;
 		*denominator = 30;
-- 
2.31.1

